src/fsharp/IlxGen.fs

```diff
@@ -4781,54 +4781,87 @@ and GenDelegateExpr cenv cgbuf eenvouter expr (TObjExprMethod((TSlotSig(_, deleg
                 false
          with _ ->
             false
+
+    // When creating a delegate to a static method, we can directly reference the method (rather than creating a closure class).
+    let staticMethodFromBody =
+        fun body ->
+            match body with
+            | Expr.App (f, _, _, _, m) ->
+                match f with
+                | Expr.Val (vref, _, _) ->
+                    let storage = StorageForValRef g m vref eenvouter
+                    match storage with
+                    | Method (_, _, mspec, _, _, _, _) when mspec.CallingConv.IsStatic ->
+                        Some(mspec)
+                    | _ -> None
+                | _ -> None
+            | _ -> None
+
+    let stripBodyExpr =
```

Comment:
probably can be written as

```fsharp
let stripBodyExpr body =
  match stripExpr body with
  | Expr.Let (_, letBody, _, _) -> letBody
  | _ -> body
```

same remark for above function, you can declare function with parameters as a let binding.

Reply:
Ooo... thanks =)